### PR TITLE
BOLT7: channel_update after exchange funding_locked

### PR DESCRIPTION
**Pull request by nayuta-ueno**
If a node sends its own `channel_update` to a peer node before receiving a `funding_lock`, the peer node may discard because it has not `short_channel_id` yet.

```
ex: node-A and node-B check blockchain every 30 seconds.

  node-A                    node-B
    |                         |
  ~~~ funding_tx is in block ~~~
    |                         |
 [detect new block]           |
 [calc short_channel_id]      |
 [create channel_update]      |
    |                         |
    |---funding_locked------->|
    |---channel_update------->|
    |                         |cannot compare short_channel_id
    |                         |  ===> discard channel_update
    |                         |
    |                         |
    |                  [detect new block]
    |                  [calc short_channel_id]
    |                  [create channel_update]
    |                         |
    |<-------funding_locked---|
    |<-------channel_update---|
compare short_channel_id      |
  ===> accept channel_update  |
    |                         |
    |                         |
```

### CODE REVIEW

**Comment by t-bast** [comment]
The spec mentions the following:

> The receiving node:
>  - if the `short_channel_id` does NOT match a previous `channel_announcement`,
>  - if the `short_channel_id` does NOT match a previous `channel_announcement`,
  OR if the channel has been closed in the meantime:
  OR if the channel has been closed in the meantime:
    - MUST ignore `channel_update`s that do NOT correspond to one of its own
    - MUST ignore `channel_update`s that do NOT correspond to one of its own
    channels.
    channels.

Don't you think this covers your point already?

**Comment by nayuta-ueno** [comment]
> Don't you think this covers your point already?
I think so.
But some node send `channel_update` before receiving `funding_locked`, and do not send `channel_update` after sending `channel_announcement`...

Should I treat this as a node bug?

**Comment by t-bast** [comment]
I think that if what you want is to prevent nodes from sending `channel_update` before receiving `funding_locked`, it would be better to add a new bullet point saying something like `MUST NOT send a channel_update before funding_locked has been received` instead of the currently proposed change. What do you think? We can discuss this point during tonight's meeting.

**Comment by nayuta-ueno** [comment]
Thank you!
I'll change it.

File: 07-routing-gossip.md [hunk 1]
```diff
@@ -445,6 +445,7 @@ The origin node:
   `announce_channel` bit was not set).
     - MUST NOT forward such a `channel_update` to other peers, for privacy
     reasons.
+    - MUST NOT send a `channel_update` before `funding_locked` has been received.
```
Comment (TheBlueMatt) [inline 07-routing-gossip.md]: Should this be clarified to only apply to such channel updates that are just destined for a channel peer?

**Comment by TheBlueMatt** [comment]
Specifically, channel_updates about a public channel are already gated as being required to come after channel_announcements.

**Comment by TheBlueMatt** [comment]
CC #536.

File: 07-routing-gossip.md (hunk 1 above)
Comment (nayuta-ueno) [reply 07-routing-gossip.md]: Thank you.

a `channel_update` ==> a created `channel_update`

**Comment by nayuta-ueno** [comment]
I think "exchange `announcement_signatures`" is same as "exchange `channel_announcement`".
So our node send a created `channel_update` to channel peer:
  - after exchange `funding_locked` (for invoice r-field)
  - after exchange `announcement_signatures` (for gossip)

File: 07-routing-gossip.md [hunk 2]
```diff
@@ -445,6 +445,7 @@ The origin node:
   `announce_channel` bit was not set).
     - MUST NOT forward such a `channel_update` to other peers, for privacy
     reasons.
+    - MUST NOT send a created `channel_update` before `funding_locked` has been received.
```
Comment (t-bast) [inline 07-routing-gossip.md]: I'm not sure this is the right place to put this. Right now this is a sub-case of when we want to create a `channel_update` _before_ announcing the channel.
I think the restriction you want to add should apply to all `channel_updates`, so it should be on a different line shouldn't it? Maybe before line 443 for example.

Comment (nayuta-ueno) [reply 07-routing-gossip.md]: Thank you.
Move to before L.443.

**Pull request merged**